lamagraph-core/src/Core/Map.hs

```diff
@@ -0,0 +1,56 @@
+module Core.Map (Map, find, insertWith) where
+
+import Clash.Prelude
+import Core.Node
+
+{- | key-value store based on `Vec`. Key is `AddressNumber`, it has linear time to find or change element.
+But iterate (such as map or fold) by value and key is easy.
+It assumed that `size` is small
+-}
+type Map (size :: Nat) v = (Vec size (Maybe (AddressNumber, Maybe v)))
+
+find ::
```

Comment:
Can it be done using [fold](https://hackage.haskell.org/package/clash-prelude-1.8.1/docs/Clash-Prelude.html#v:fold)? It should be more efficient. I hope.